### DISCUSSION TITLE
Investigation: Does sdc11073 send WS-Discovery Probe messages without specifying an encoding?

### DISCUSSION THREAD

**Discussion by ben-Draeger** [Draegerwerk/sdc11073#299] [General]
When starting an SDC discovery and capturing the traffic using Wireshark, I find WS-Discovery Probe messages like the following

```xml
<s12:Envelope 
  xmlns:wsd="http://docs.oasis-open.org/ws-dd/ns/discovery/2009/01" 
  xmlns:wsa="http://www.w3.org/2005/08/addressing" 
  xmlns:s12="http://www.w3.org/2003/05/soap-envelope" 
  xmlns:dpws="http://docs.oasis-open.org/ws-dd/ns/dpws/2009/01">
  <s12:Header>
    <wsa:Action>http://docs.oasis-open.org/ws-dd/ns/discovery/2009/01/Probe</wsa:Action>
	<wsa:MessageID>urn:uuid:27f584fe-c64e-4f4b-a10a-0b561acf625f</wsa:MessageID>
	<wsa:To>urn:docs-oasis-open-org:ws-dd:ns:discovery:2009:01</wsa:To>
  </s12:Header>
  <s12:Body>
    <wsd:Probe>
	  <wsd:Types 
	    xmlns:CEtSOGHRet="http://docs.oasis-open.org/ws-dd/ns/dpws/2009/01" 
		xmlns="http://docs.oasis-open.org/ws-dd/ns/dpws/2009/01" 
		xmlns:iFyBUSdTqv="http://standards.ieee.org/downloads/11073/11073-20702-2016">Device iFyBUSdTqv:MedicalDevice</wsd:Types>
	</wsd:Probe>
  </s12:Body>
</s12:Envelope>
```

which do not specify an encoding (WS-Discovery Probe Messages are send via UDP Multicast, so the only place they can specify an encoding at is an XML declaration. However, as you can see, the above message does not have an XML declaration).

Without a specified encoding, the only way to determine the encoding of the above message is XML encoding autodetection, a rather complicated process specified in [Appendix F.1 of the XML standard](https://www.w3.org/TR/xml/#sec-guessing-no-ext-info). In order to maximize Interoperability it is not wise to rely upon all SDC devices correctly implementing this complicated process. I would hence strongly suggest to add an XML declaration and to specify the encoding.

**Comment by deichmab-draeger** [comment]
I created an issue.

**Reply by leon1995** [reply]
the issue created is #314

**Comment by leon1995** [comment]
The issue has been merged. I close this discussion. If the problem has not been solved to your satisfaction, then just reopen the discussion
